### PR TITLE
[TACHYON-957] Collect metrics for the new client and master

### PR DESCRIPTION
**Pull request by carsonwang**
https://tachyon.atlassian.net/browse/TACHYON-957
Add the metrics collection for the new client and master.

### CODE REVIEW

File: clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java [hunk 1]
```diff
@@ -60,6 +66,12 @@ public void close() {
 
   @Override
   public void release(WorkerClient workerClient) {
+    try {
+      // Heartbeat to send the client metrics.
+      workerClient.sessionHeartbeat();
+    } catch (IOException ioe) {
+      LOG.warn("Failed sending client metrics before releasing the worker client", ioe);
+    }
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java]: Do the metrics need to be cleared here?

**Comment by calvinjia** [comment]
@carsonwang Thanks for adding this back, just one minor comment.

File: clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java (hunk 1 above)
Comment (carsonwang) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java]: No, we don't need to clear the metrics here.  All the worker clients share a single client metrics object stored in the client context. The metrics will be cleared when doing the heartbeat. Even if an exception happens, the metrics that are not cleared will be sent by the next heartbeat.

**Comment by gpang** [comment]
Thanks for this patch @carsonwang ! The master now has a MasterContext similar to the client, so I think you can just put the metrics in there, and avoid passing it to everything. Thanks!

File: clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java (hunk 1 above)
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java]: Got it, thanks.

**Comment by calvinjia** [comment]
@carsonwang This looks good to me aside from the comment about the master context.

**Comment by carsonwang** [comment]
@calvinjia , I have moved the master metrics to the master context.

**Comment by calvinjia** [comment]
Jenkins, test this please.

File: servers/src/test/java/tachyon/master/file/FileSystemMasterTest.java [hunk 2]
```diff
@@ -64,7 +65,8 @@ public void before() throws Exception {
     Journal fsJournal = new Journal(mTestFolder.newFolder().getAbsolutePath(), mTachyonConf);
 
     mBlockMaster = new BlockMaster(mTachyonConf, blockJournal);
-    mFileSystemMaster = new FileSystemMaster(mTachyonConf, mBlockMaster, fsJournal);
+    mFileSystemMaster =
+        new FileSystemMaster(mTachyonConf, mBlockMaster, fsJournal);
```
Comment (calvinjia) [inline servers/src/test/java/tachyon/master/file/FileSystemMasterTest.java]: Could you revert the formatting here?

File: servers/src/main/java/tachyon/master/TachyonMaster.java [hunk 3]
```diff
@@ -162,9 +165,12 @@ protected TachyonMaster() {
       mBlockMaster = new BlockMaster(mTachyonConf, mBlockMasterJournal);
       mFileSystemMaster =
           new FileSystemMaster(mTachyonConf, mBlockMaster, mFileSystemMasterJournal);
-      mRawTableMaster = new RawTableMaster(mTachyonConf, mFileSystemMaster, mRawTableMasterJournal);
+      mRawTableMaster =
+          new RawTableMaster(mTachyonConf, mFileSystemMaster, mRawTableMasterJournal);
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/master/TachyonMaster.java]: Could you revert the formatting here?

File: integration-tests/src/test/java/tachyon/master/MasterTestUtils.java [hunk 4]
```diff
@@ -33,7 +33,8 @@ public static FileSystemMaster createFileSystemMasterFromJournal(TachyonConf tac
         tachyonConf);
 
     BlockMaster blockMaster = new BlockMaster(tachyonConf, blockJournal);
-    FileSystemMaster fsMaster = new FileSystemMaster(tachyonConf, blockMaster, fsJournal);
+    FileSystemMaster fsMaster =
+        new FileSystemMaster(tachyonConf, blockMaster, fsJournal);
```
Comment (calvinjia) [inline integration-tests/src/test/java/tachyon/master/MasterTestUtils.java]: Could you revert the formatting here?

**Comment by calvinjia** [comment]
@carsonwang just a few minor formatting issues, could you also merge master?

**Comment by gpang** [comment]
After resolving the merge conflicts, looks good. Thanks for the feature, @carsonwang !

**Comment by carsonwang** [comment]
Merge conflicts and format issue are resolved. @calvinjia @gpang

**Comment by carsonwang** [comment]
Jenkins, test this please.

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
@carsonwang Could you resolve the conflicts with master please? Also it seems like the tests are not happy on Jenkins, do they pass locally?

**Comment by carsonwang** [comment]
@calvinjia  conflicts resolved. Tests passed now.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**